Src/Sarif.PatternMatcher/SearchSkimmer.cs

```diff
@@ -209,25 +208,28 @@ public override void Analyze(AnalyzeContext context)
                     continue;
                 }
 
-                if (matchExpression.MatchLengthToDecode > 0)
+                if (matchExpression.Base64EncodingMatch?.IsValid() == true)
                 {
-                    decimal unencodedLength = matchExpression.MatchLengthToDecode;
-
                     // Every 3 bytes of a base64-encoded string produces 4 bytes of data.
-                    int unpaddedLength = (int)Math.Ceiling(decimal.Divide(unencodedLength * 8M, 6M));
-                    int paddedLength = 4 * (int)Math.Ceiling(decimal.Divide(unencodedLength, 3M));
+                    int unpaddedMinLength = (int)Math.Ceiling(decimal.Divide(matchExpression.Base64EncodingMatch.MinSourceLength * 8M, 6M));
+                    int unpaddedMaxLength = (int)Math.Ceiling(decimal.Divide(matchExpression.Base64EncodingMatch.MaxSourceLength * 8M, 6M));
 
-                    // Create proper regex for base64-encoded string which includes padding characters.
+                    // Create proper regex for all strings matches base64 charset and length requirements
+                    // But these strings may be not valid base64 strings since the padding length are vary
+                    // for different length source string.
                     string base64DecodingRegexText =
```

Comment:
>base64DecodingRegexText [](http://example.com/codeflow?start=27&length=23)

Couldn't figure out a single regex pattern can match all base64 encoded string of a specific length range, because the padding pattern varies.

Have to first detect the possible base64 string match the length requirement, then check if they are valid base64 strings.

let me know if any better idea

Reply:
@yongyan-gh Trying to understand this better. Can we not parameterize the regular expression based on the values specified in the range of `Base64EncodingMatch`?

Reply:
@suvamM can you pls elaborate more about your idea? do you mean ignore the length check and find all valid base64 strings?